src/algebra/jordan/basic.lean

```diff
@@ -0,0 +1,223 @@
+  is_comm_jordan.jordan, is_comm_jordan.mul_comm b (a * a)],
+  lmul_comm_rmul_rmul := λ a b, by rw [is_comm_jordan.mul_comm, ←is_comm_jordan.jordan,
+    is_comm_jordan.mul_comm],
+  lmul_lmul_comm_rmul :=  λ a b, by rw [is_comm_jordan.mul_comm (a * a) (b * a),
+    is_comm_jordan.mul_comm b a, is_comm_jordan.jordan, is_comm_jordan.mul_comm,
+    is_comm_jordan.mul_comm b (a * a)],
+  rmul_comm_rmul_rmul := λ a b, by rw [is_comm_jordan.mul_comm b a, is_comm_jordan.jordan,
+    is_comm_jordan.mul_comm], }
+
+universe u
+
+/-- Semigroup multiplication satisfies the (non-commutative) Jordan axioms-/
+@[priority 100] -- see Note [lower instance priority]
+instance semigroup.is_jordan (B : Type u) [semigroup B] : is_jordan B :=
+{ lmul_comm_rmul := λ a b, by rw mul_assoc,
+  lmul_lmul_comm_lmul := λ a b, by rw [mul_assoc, mul_assoc],
+  lmul_comm_rmul_rmul := λ a b, by rw [mul_assoc],
+  lmul_lmul_comm_rmul := λ a b, by rw [←mul_assoc],
+  rmul_comm_rmul_rmul := λ a b, by rw [← mul_assoc, ← mul_assoc], }
+
+@[priority 100] -- see Note [lower instance priority]
+instance comm_semigroup.is_comm_jordan (B : Type u) [comm_semigroup B] : is_comm_jordan B :=
+{ mul_comm := mul_comm,
+  jordan := λ a b, mul_assoc _ _ _, }
+
+variables {A : Type*}
+
+namespace add_monoid.End
+variables [non_unital_non_assoc_semiring A]
+
+/-- Left multiplication operator. This is a variant of `add_monoid_hom.mul_left`. -/
+@[simps] def L : A →+ add_monoid.End A := add_monoid_hom.mul
+
+/-- Right multiplication operator. This is a variant of `add_monoid_hom.mul_right`. -/
+@[simps] def R : A →+ add_monoid.End A := (L : A →+ add_monoid.End A).flip
+
+end add_monoid.End
+
+open add_monoid.End (L R)
+
+/-! The Jordan axioms can be expressed in terms of commuting multiplication operators -/
+section lie
+variables [non_unital_non_assoc_ring A] [is_jordan A]
+
+@[simp] lemma lie_L_R (a : A) : ⁅L a, R a⁆ = 0 :=
+add_monoid_hom.ext $ λ b, sub_eq_zero_of_eq (is_jordan.lmul_comm_rmul _ _).symm
+
+@[simp] lemma lie_L_L_sq (a : A) : ⁅L a, L (a * a)⁆ = 0 :=
+add_monoid_hom.ext $ λ b, sub_eq_zero_of_eq (is_jordan.lmul_lmul_comm_lmul _ _).symm
+
+@[simp] lemma lie_L_R_sq (a : A) : ⁅L a, R (a * a)⁆ = 0 :=
+add_monoid_hom.ext $ λ b, sub_eq_zero_of_eq (is_jordan.lmul_comm_rmul_rmul _ _).symm
+
+@[simp] lemma lie_L_sq_R (a : A) : ⁅L (a * a), R a⁆ = 0 :=
+add_monoid_hom.ext $ λ b, sub_eq_zero_of_eq (is_jordan.lmul_lmul_comm_rmul _ _)
+
+@[simp] lemma lie_R_R_sq (a : A) : ⁅R a, R (a * a)⁆ = 0 :=
+add_monoid_hom.ext $ λ b, sub_eq_zero_of_eq (is_jordan.rmul_comm_rmul_rmul _ _).symm
+
+end lie
+
+variables [non_unital_non_assoc_ring A] [is_comm_jordan A]
+
+/-
+instance : comm_monoid A :=
+{ mul_comm := λ a b, is_comm_jordan.mul_comm a b,
+  .. (show non_unital_non_assoc_ring A, by apply_instance) }
+-/
+
+/-- Linearise the Jordan axiom with two variables -/
+lemma mul_op_com1 (a b : A) :
+  ⁅L a, L (b*b)⁆ + ⁅L b, L (a*a)⁆ + (2:ℤ)•⁅L a, L (a*b)⁆ + (2:ℤ)•⁅L b, L (a*b)⁆ = 0 :=
+begin
```

Comment:
This lemma name and docstring don't do nearly enough to explain what is going on with this statement. Can you add a sentence or two explaining what the idea here is?

Reply:
I've pushed a commit which expands on the idea of the lemma and renames it.